### PR TITLE
Support query events stacked on incremental_load events in summarize.

### PR DESCRIPTION
**Pull request by michaelwoerister**
Fixes the issue found in https://github.com/rust-lang/rust/pull/66981. The should be correct but this could be cleaned up even more if the questions in https://github.com/rust-lang/measureme/issues/75 were properly answered. Would be good to do that before the next bigger release.

r? @wesleywiser or @andjo403

### CODE REVIEW

File: summarize/src/analysis.rs [hunk 1]
```diff
@@ -159,33 +159,47 @@ pub fn perform_analysis(data: ProfilingData) -> Results {
...
                     record_event_data(&current_top.label, &|data| {
-                        data.self_time -= current_event.duration().unwrap();
+                        match &current_top.event_kind[..] {
+                            QUERY_EVENT_KIND | GENERIC_ACTIVITY_EVENT_KIND => {
+                                data.self_time -= current_event_duration;
+                            }
+                            INCREMENTAL_LOAD_RESULT_EVENT_KIND => {
+                                data.incremental_load_time  -= current_event_duration;
+                            }
+                            _ => {
```
Comment (wesleywiser) [inline summarize/src/analysis.rs]: We probably need to handle `QUERY_BLOCKED_EVENT_KIND` here.

Comment (michaelwoerister) [reply summarize/src/analysis.rs]: Hm, that definitely shouldn't have anything on top of it, right?

Comment (wesleywiser) [reply summarize/src/analysis.rs]: I'm not sure I know enough about how the query system works when parallelism is turned on to say. Does the thread stay blocked when a query is blocked or does it try to work on another query?

Comment (michaelwoerister) [reply summarize/src/analysis.rs]: It stays blocked.

Comment (wesleywiser) [reply summarize/src/analysis.rs]: Ah ok. You're right then.

**Pull request merged**